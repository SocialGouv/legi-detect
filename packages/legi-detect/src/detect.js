const { detectSingle, detectMultiples } = require("./detect.articles");

const detect = (str, defaultCode) => {
  return [
    ...detectSingle(str, defaultCode),
    ...detectMultiples(str, defaultCode)
  ];
};

module.exports = detect;
