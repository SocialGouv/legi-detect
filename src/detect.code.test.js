import { detectCode } from "./detect.code";

const tests = `
du code civil et tout
code civille
code civil
code pénal
code penal
du code pennal
du code penal
du code pénal et du code civil et du code de la route
du code de la route
du code de la rue
`
  .split("\n")
  .filter(Boolean);

tests.forEach(test => {
  it(test, () => {
    expect(detectCode(test)).toMatchSnapshot();
  });
});
