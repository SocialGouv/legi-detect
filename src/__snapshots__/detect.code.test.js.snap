// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code civil 1`] = `
Object {
  "id": "LEGITEXT000006070721",
  "source": "code civil",
  "value": "Code civil",
}
`;

exports[`code civille 1`] = `undefined`;

exports[`code penal 1`] = `
Object {
  "id": "LEGITEXT000006071029",
  "source": "code penal",
  "value": "CODE PENAL",
}
`;

exports[`code pénal 1`] = `
Object {
  "id": "LEGITEXT000006070719",
  "source": "code pénal",
  "value": "Code pénal",
}
`;

exports[`du code civil et tout 1`] = `
Object {
  "id": "LEGITEXT000006070721",
  "source": "code civil",
  "value": "Code civil",
}
`;

exports[`du code de la route 1`] = `
Object {
  "id": "LEGITEXT000006074947",
  "source": "code de la route",
  "value": "Code de la route",
}
`;

exports[`du code de la rue 1`] = `undefined`;

exports[`du code penal 1`] = `
Object {
  "id": "LEGITEXT000006071029",
  "source": "code penal",
  "value": "CODE PENAL",
}
`;

exports[`du code pennal 1`] = `
Object {
  "id": "LEGITEXT000006071029",
  "source": "code pennal",
  "value": "CODE PENAL",
}
`;

exports[`du code pénal et du code civil et du code de la route 1`] = `
Object {
  "id": "LEGITEXT000006070719",
  "source": "code pénal",
  "value": "Code pénal",
}
`;
